sdk/identity/azure-identity/azure/identity/_credentials/auth_file.py

```diff
@@ -0,0 +1,48 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import six
+import json
+
+from .client_credential import ClientSecretCredential
+from azure.core.exceptions import ClientAuthenticationError
+
+
+def _parse_credentials_file(file_path):
+    with open(file_path, 'rb') as f:
+        auth_str = six.ensure_str(f.read())
+        
+    return json.loads(auth_str)
+
+
+class AuthFileCredential(object):
+
+    def __init__(self, file_path, **kwargs):
+        self._credential = None
+        self.file_path = file_path
+        self.kwargs = kwargs
```

Comment:
Suggesting copying this to prevent mutation, and marking it private (e.g. `_init_kwargs`).